tests/llms/test_cohere.py

```diff
@@ -0,0 +1,149 @@
+    return cohere.responses.Generations.from_dict(
+        {
+            "id": "21caa4c4-6b88-45f7-b144-14ef4985384c",
+            "generations": [
+                {
+                    "id": "b5e2bb70-bc9c-4f86-a22e-5b5fd13a3482",
+                    "text": "\n\nThis is indeed a test",
+                    "finish_reason": "COMPLETE",
+                }
+            ],
+            "prompt": "test prompt",
+            "meta": {"api_version": {"version": "1"}},
+        },
+        return_likelihoods=False,
+    )
+
+
+def mock_chat_with_retry(*args: Any, **kwargs: Any) -> dict:
+    return cohere.responses.Chat.from_dict(
+        {
+            "chatlog": None,
+            "citations": None,
+            "conversation_id": None,
+            "documents": None,
+            "generation_id": "357d15b3-9bd4-4170-9439-2e4cef2242c8",
+            "id": "25c3632f-2d2a-4e15-acbd-804b976d0568",
+            "is_search_required": None,
+            "message": "test prompt",
+            "meta": {"api_version": {"version": "1"}},
+            "preamble": None,
+            "prompt": None,
+            "response_id": "25c3632f-2d2a-4e15-acbd-804b976d0568",
+            "search_queries": None,
+            "search_results": None,
+            "text": "\n\nThis is indeed a test",
+            "token_count": {
+                "billed_tokens": 66,
+                "prompt_tokens": 64,
+                "response_tokens": 9,
+                "total_tokens": 73,
+            },
+        },
+        client=None,
+        message="test_prompt",
+    )
+
+
+async def mock_achat_with_retry(*args: Any, **kwargs: Any) -> dict:
+    return cohere.responses.Chat.from_dict(
+        {
+            "chatlog": None,
+            "citations": None,
+            "conversation_id": None,
+            "documents": None,
+            "generation_id": "357d15b3-9bd4-4170-9439-2e4cef2242c8",
+            "id": "25c3632f-2d2a-4e15-acbd-804b976d0568",
+            "is_search_required": None,
+            "message": "test prompt",
+            "meta": {"api_version": {"version": "1"}},
+            "preamble": None,
+            "prompt": None,
+            "response_id": "25c3632f-2d2a-4e15-acbd-804b976d0568",
+            "search_queries": None,
+            "search_results": None,
+            "text": "\n\nThis is indeed a test",
+            "token_count": {
+                "billed_tokens": 66,
+                "prompt_tokens": 64,
+                "response_tokens": 9,
+                "total_tokens": 73,
+            },
+        },
+        client=None,
+        message="test_prompt",
+    )
+
+
+@pytest.mark.skipif(cohere is None, reason="cohere not installed")
+def test_completion_model_basic(monkeypatch: MonkeyPatch) -> None:
+    monkeypatch.setattr(
+        "llama_index.llms.cohere.completion_with_retry", mock_completion_with_retry
+    )
+    mock_api_key = "5secFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF"
```

Comment:
Just wondering, why choose this key?  Is it to match a length requirement?  I like to just use `"fake"`, that way it's clear the key isn't a real key

Reply:
There isn't a length requirement, using "fake" probably makes more sense here. I'll update the PR with that.  My reasoning was, use the first few characters of a real key and then pad it with F's to match the length of a real key to give anyone looking at the tests an idea of what a api key looks like here.

Reply:
Oh yeah maybe then name it `"fake_with_40_charsssssssssssssssss"` or something like that, lol